refactor: replace breadcrumbs_on_rails with custom breadcrumbs implementation

katafrakt opened the pull request:
# Description
<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

`breadcrumbs_on_rails` gem caused incompatibility with Rails projects which used other breadcrumbs gems using same method names. Instead, an internal implementation based on `breadcrumbs_on_rails` was created, using method names with "avo" (`add_avo_breadcrumb`, `avo_breadcrumbs`...).

The implementation was also stripped down to only use features that Avo uses.

### Questions

* This does not add any tests on its own. I'm willing to write some, but there's not any place where it fits. Should I create `spec/concerns`?
* Currently `lib/avo/concerns` only contains model concerns (I think), so not sure if it's okay to mix in a controller concern in there.

Fixes #1394 

# Checklist:
<!--
  Please go through the steps and complete them if they make sense (add tests if the change requires it, add to the docs, etc.)
  (Mark [x] inside the brackets)
-->

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/docs)
- [ ] I have added tests that prove my fix is effective or that my feature works


## Manual review steps
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

1. In example app, go to any subsection (e.g. `/admin/resources/products`)
2. Verify that breadcrumbs display correctly and are clickable

Manual reviewer: please leave a comment with output from the test if that's the case.

katafrakt commented:
I should probably also mentioned that I tried this out with a problematic app from #1394 and it no longer conflicts with the other gem.

adrianthedev commented:
Hey @katafrakt. I'm curious if we **need** to rename the `add_breadcrumb` method to `add_avo_breadcrumb`.

This is a breaking change for all apps as it's present in the default avo.rb initializer.

adrianthedev commented:
Also, I see this issue in a manual test.

![CleanShot 2022-11-08 at 13 55 26](https://user-images.githubusercontent.com/1334409/200557841-e6044253-67cc-42fb-a8ff-a9ee48bd0ef4.gif)

katafrakt commented:
Thanks for checking @adrianthedev!

There was indeed a bug with appending breadcrumb on top of previous ones - and I think I fixed that.

As for using `add_avo_breadcrumbs`, good point with initializer and breaking changes. I checked a few existing breadcrumbs implementations for Rails and they are not using `add_breadcrumb` method (except, of course, `breadcrumbs_on_rails`, but the implementation should be compatible). After that I reverted the name of the method back to `add_breadcrumb`.

adrianthedev commented:
That's perfect! Thanks for understanding @katafrakt, and for making the changes back 🙏
